module/zfs/zfs_ioctl.c

```diff
@@ -4201,29 +4201,50 @@ zfs_ioc_recv_impl(char *tofs, char *tosnap, char *origin,
 			 * regardless.
 			 */
 			if (!first_recvd_props)
-				props_reduce(props, origprops);
+				props_reduce(recvprops, origprops);
 			if (zfs_check_clearable(tofs, origprops, &errlist) != 0)
 				(void) nvlist_merge(*errors, errlist, 0);
 			nvlist_free(errlist);
 
 			if (clear_received_props(tofs, origprops,
-			    first_recvd_props ? NULL : props) != 0)
+			    first_recvd_props ? NULL : recvprops) != 0)
 				*errflags |= ZPROP_ERR_NOCLEAR;
 		} else {
 			*errflags |= ZPROP_ERR_NOCLEAR;
 		}
 	}
 
-	if (props != NULL) {
+	if (recvprops != NULL) {
 		props_error = dsl_prop_set_hasrecvd(tofs);
 
 		if (props_error == 0) {
-			delayprops = extract_delay_props(props);
+			delayprops = extract_delay_props(recvprops);
 			(void) zfs_set_prop_nvlist(tofs, ZPROP_SRC_RECEIVED,
-			    props, *errors);
+			    recvprops, *errors);
 		}
 	}
 
+	if (cmdprops != NULL) {
+		nvlist_t *oprops = fnvlist_alloc();
+		// nvlist_t *xprops = fnvlist_alloc();
+		nvpair_t *nvp = NULL;
+		while ((nvp = nvlist_next_nvpair(cmdprops, nvp)) != NULL) {
+			if (nvpair_type(nvp) == DATA_TYPE_BOOLEAN) {
+				const char *name = nvpair_name(nvp);
+				// fnvlist_add_string(xprops, name, "");
+				// fnvlist_add_boolean(xprops, name);
+				dsl_prop_inherit(tofs, name,
+				    ZPROP_SRC_INHERITED);
```

Comment:
Kind of unfortunate performance characteristics here, if there are a bunch of `-x` options, since this does a `txg_wait_synced(0)`.  But I suppose it's OK for now.